tmk_core/common/action_tapping.c

```diff
@@ -100,20 +111,18 @@ bool process_tapping(keyrecord_t *keyp)
                     // enqueue
                     return false;
                 }
-#if TAPPING_TERM >= 500 || defined PERMISSIVE_HOLD
                 /* Process a key typed within TAPPING_TERM
                  * This can register the key before settlement of tapping,
                  * useful for long TAPPING_TERM but may prevent fast typing.
                  */
-                else if (IS_RELEASED(event) && waiting_buffer_typed(event)) {
+                else if ( ( permissive_hold || get_tapping_term(tapping_key.event) >= 500) && IS_RELEASED(event) && waiting_buffer_typed(event)) {
```

Comment:
You could do this instead of defining `bool permissive_hold`. Should be functionally equivalent.
```suggestion
#ifndef PERMISSIVE_HOLD
                else if (get_tapping_term(tapping_key.event) >= 500 && IS_RELEASED(event) && waiting_buffer_typed(event))
#endif
                {
```

Reply:
I'm not sure this is correct. 

To be clear, the original block is:
```c
#if TAPPING_TERM >= 500 || defined PERMISSIVE_HOLD
                /* Process a key typed within TAPPING_TERM
                 * This can register the key before settlement of tapping,
                 * useful for long TAPPING_TERM but may prevent fast typing.
                 */
                else if (IS_RELEASED(event) && waiting_buffer_typed(event)) {
                    debug("Tapping: End. No tap. Interfered by typing key\n");
                    process_record(&tapping_key);
                    tapping_key = (keyrecord_t){};
                    debug_tapping_key();
                    // enqueue
                    return false;
                }
#endif
```

However, I want to make sure that this correctly checks the tapping term for the specific key in question, not the global value (since they may not be the same. 

In theory, I want this:
```
#if get_tapping_term(tapping_key.event)>= 500 || defined PERMISSIVE_HOLD
                /* Process a key typed within TAPPING_TERM
                 * This can register the key before settlement of tapping,
                 * useful for long TAPPING_TERM but may prevent fast typing.
                 */
                else if (IS_RELEASED(event) && waiting_buffer_typed(event)) {
                    debug("Tapping: End. No tap. Interfered by typing key\n");
                    process_record(&tapping_key);
                    tapping_key = (keyrecord_t){};
                    debug_tapping_key();
                    // enqueue
                    return false;
                }
#endif
```

But between the tiredness I was feeling, and unfamiliarity with programming.... 


Though, I'm seriously considering making this stuff optional, as it adds a couple hundred bytes to the firmware size on my boards (lto is enabled, so probably more), since it's no longer preprocessed out.